src/include/daos_srv/vos.h

```diff
@@ -1034,6 +1034,21 @@ vos_iter_process(daos_handle_t ih, vos_iter_proc_op_t op, void *args);
 int
 vos_iter_empty(daos_handle_t ih);
 
+/**
+ * When the callback executes code that may yield, in some cases, it needs to
+ * verify the value or key it is operating on still exists before continuing.
+ * This function will revalidate from the object layer down.   This is only
+ * supported in conjunction with recursive vos_iterate from VOS_ITER_OBJ.
```

Comment:
It looks to me an issue existing in EC aggregation & rebuild enumeration as well: when EC aggregation ULT or rebuild enumeration ULT yield in iter callback, and VOS aggregation deleted it's parent, EC aggregation or rebuild enumeration will resume to work in an deleted sub-tree, right?

I'm wondering if we need some sort of refcount mechanism to avoid the race of sub-tree deletion and sub-tree iteration (which could yield)?

Reply:
I don't think it can happen with EC aggregation because it operates on a different epoch range.  We can't do VOS aggregation on a given range until EC aggregation is complete so I don't think we can ever hit this situation where we delete something that is being operated on.

Rebuild might be different.  @wangdi1 said that it's also not possible but I'm not entirely convinced because unlike EC aggregation, it's working at a lower epoch range than vos aggregation.